internal/playlist_creator/DirectoryReader.go

```diff
@@ -0,0 +1,64 @@
+package playlist_creator
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"playlistCreator/internal/playlist_creator/config"
+	"sort"
+	"strings"
+)
+
+func ReadFiles(config *config.Config) *FileData {
+	var fileData = FileData{}
+
+	readFilesRecursively(config, config.InputPath, &fileData)
+
+	sort.Slice(fileData.FilesList, func(i, j int) bool {
+		return fileData.FilesList[i].FileName < fileData.FilesList[j].FileName
+	})
+
+	return &fileData
+}
+
+func readFilesRecursively(config *config.Config, directoryNameAndPath string, fileData *FileData) {
+	files, err := os.ReadDir(directoryNameAndPath)
+	if err != nil {
+		log.Printf("Warning: Could not read directory %s: %s", directoryNameAndPath, err.Error())
+	}
+
+	// https://golang.cafe/blog/how-to-list-files-in-a-directory-in-go.html
+	// filepath.Walk
+	for _, file := range files {
+		if !file.IsDir() {
+			var extension = filepath.Ext(file.Name())
```

Comment:
```suggestion
		 extension := filepath.Ext(file.Name())
```

I can't find the link now, but it's recommended that if you are declaring a variable outside a block, use `var` otherwise use `:=`